sycl/tools/sycl-xocc/bin/sycl-xocc

```diff
@@ -153,45 +173,50 @@ else
   usage 14 "the path given for the temporary directory is an invalid directory"
 fi
 
-# excluding file extension e.g. .cpp and path
-SOURCE_FILE_NAME="$3"
-# input file name including extensions and path
-INPUT_FILE_NAME="$4"
 # Should point to whatever Clang has decided is the temporary directory
-TMP_PATH="$(mktemp $5/sycl-xocc.XXXXXX)"
+TMP_PATH="$(mktemp ${TMP_DIR}/sycl-xocc.XXXXXX)"
+
 
-# output file name including extensions and path
-OUTPUT_FILE_NAME="$6"
 
 set +e
 # allow faillures because empty string is good default
 # This make debugging easier.
-ADDITIONAL_COMPILE_ARGS=$(cat $7)
-ADDITIONAL_LINK_ARGS=$(cat $8)
+ADDITIONAL_COMPILE_ARGS=$(cat ${ADDITIONAL_COMPILE_FILE})
+ADDITIONAL_LINK_ARGS=$(cat ${ADDITIONAL_LINK_FILE})
 set -e
-XCL_EMULATION_MODE=$9
 
 KERNELPROP="${TMP_PATH}_KernelProperties_${SOURCE_FILE_NAME}.bash"
 
 cp "$INPUT_FILE_NAME" "${TMP_PATH}_${SOURCE_FILE_NAME}_kernels.bc"
 
-# -flat-address-space defines the generic address space we wish to deduce to
-# something concrete, used by the infer-address-spaces pass which is slightly
-# modified from the upstream implementation to take this argument to avoid
-# adding a new LLVM Target that will largelly be unused right now..
-$OPT --sycl-xocc -preparesycl -globaldce -inline -infer-address-spaces -globaldce -O3 -globaldce\
-  -flat-address-space=0 -globaldce  -globaldce -kernelPropGen \
-  --sycl-kernel-propgen-output $KERNELPROP \
-  #-inSPIRation \
-  "${TMP_PATH}_${SOURCE_FILE_NAME}_kernels.bc" -o "${TMP_PATH}_${SOURCE_FILE_NAME}_kernels-optimized.bc"
-
-# Link our SDx SPIR binaries in
-$LLVM_LINK "${TMP_PATH}_${SOURCE_FILE_NAME}_kernels-optimized.bc" \
-  --only-needed $SDX_LIB_SPIR -o "${TMP_PATH}_${SOURCE_FILE_NAME}_kernels-linked.xpirbc"
+if [[ "${XLX_COMPILE_FLOW}" == "hls" ]]; then
+  echo "Using HLS flow"
+  $OPT --sycl-xocc -preparesycl -globaldce -O3 -globaldce -globaldce -kernelPropGen \
+    --sycl-kernel-propgen-output $KERNELPROP \
+    -inSPIRation \
+    "${TMP_PATH}_${SOURCE_FILE_NAME}_kernels.bc" -o "${TMP_PATH}_${SOURCE_FILE_NAME}_kernels-linked.xpirbc"
+  EXTRA_LD_ARGS='--sycl-xlx-hls'
+else
+  # -flat-address-space defines the generic address space we wish to deduce to
+  # something concrete, used by the infer-address-spaces pass which is slightly
+  # modified from the upstream implementation to take this argument to avoid
+  # adding a new LLVM Target that will largelly be unused right now..
+  $OPT --sycl-xocc -preparesycl -globaldce -inline -infer-address-spaces -globaldce -O3 -globaldce\
+    -flat-address-space=0 -globaldce  -globaldce -kernelPropGen \
+    --sycl-kernel-propgen-output $KERNELPROP \
+    --sycl-xlx-oclmd -inSPIRation \
+    "${TMP_PATH}_${SOURCE_FILE_NAME}_kernels.bc" -o "${TMP_PATH}_${SOURCE_FILE_NAME}_kernels-optimized.bc"
+
+  # Link our SDx SPIR binaries in
+  $LLVM_LINK "${TMP_PATH}_${SOURCE_FILE_NAME}_kernels-optimized.bc" \
+    --only-needed $SDX_LIB_SPIR -o "${TMP_PATH}_${SOURCE_FILE_NAME}_kernels-linked.xpirbc"
```

Comment:
We still needs the SPIR library? I guess there are also some address spaces uses in the SPIR library too...

Reply:
As of now , if I recall correctly the discussion we had with @Ralender , the frontend convert libm calls to spir equivalent calls, which are in turn mapped to HLS compatible constructs with the linkage of this library.

There was no address space issue when compiling a test application that uses mathematical function with the HLS flow.